config/crd/standard/gateway.networking.k8s.io_gatewayclasses.yaml

```diff
@@ -78,6 +78,9 @@ spec:
                 minLength: 1
                 pattern: ^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*\/[A-Za-z0-9\/\-._~%!$&'()*+,;=:]+$
                 type: string
+                x-kubernetes-validations:
```

Comment:
Gateway API already promises to support the [trailing 5 k8s versions](https://gateway-api.sigs.k8s.io/concepts/versioning/#supported-versions), so I believe we'd need to release a second set of CRDs for k8s <=1.24 as long as we support those versions. Based on the k8s release cycle, I think that's going to be at least another ~6 months. 

It is annoying to maintain yet another set of CRDs, but on the other hand, I think a significant amount of usage is moving towards 1.25+, and there's a chance we may be able to entirely remove the webhook on those versions in favor of CEL, which would be a significant win.

Reply:
I wish we could get away with saying v1.25 is the minimum supported K8s version for GA.

Reply:
Before we commit to double CRDs, can someone confirm its not gracefully handled on old versions? I assume not, but worth checking

Reply:
Thanks for bringing this up!

Yes so I did check things with 1.23 and I didn't find any problem. To be specific, I was able to create these CRDs which have this new `x-kubernetes-validations` field. It got ignored.

Reply:
@shaneutt at a minimum I'd like to make CEL the default installation method and then add a note like "if you're running Kubernetes 1.24 or older, installing the webhook is recommended as well".

@gauravkghildiyal that's great that we might be able to continue to use the same sets of CRDs for this!

Reply:
Chatted with @jpbetz, it looks like CRDs with `x-kubernetes-validations` will go through starting in 1.23 thanks to https://github.com/kubernetes/kubernetes/pull/108030. The actual validation will be limited to 1.25+ (unless clusters have had CEL manually enabled earlier).

Reply:
So that means that once 1.28 is out, we will be both mandating an minimum of 1.23 for the CRDs that include `x-kubernetes-validations` and still in compliance with our "5 versions" support policy.

Including another set of CRDs for older versions seems like it would be more work that it's worth then.